build/evergreen_config_lib/variants.py

```diff
@@ -245,20 +253,26 @@ def days(n):
             ['release-compile',
             'debug-compile-nosasl-nossl',
              '.debug-compile !.sspi .openssl',
+             '.debug-compile !.sspi .openssl-static',
              '.debug-compile !.sspi .nossl',
              '.latest .openssl !.nosasl .server',
+             '.latest .openssl-static !.nosasl .server',
              '.latest .nossl',
              '.4.4 .openssl !.nosasl .server',
-             '.4.2 .openssl !.nosasl .server'],
+             '.4.4 .openssl-static !.nosasl .server',
+             '.4.2 .openssl !.nosasl .server',
+             '.4.2 .openssl-static !.nosasl .server'],
```

Comment:
IMO we do not need to test more than one server version with `openssl-static`. How we link against OpenSSL should have no bearing on the behavior. It seems worthwhile to have one live test as a safety check, so I propose we only test with the latest server version.